driver-core/src/main/com/mongodb/internal/connection/LoadBalancedCluster.java

```diff
@@ -0,0 +1,230 @@
+        }
+    }
+
+    private void init(final ClusterId clusterId, final ClusterableServerFactory serverFactory, final ServerAddress host) {
+        description = new ClusterDescription(ClusterConnectionMode.LOAD_BALANCED, ClusterType.LOAD_BALANCED,
+                singletonList(ServerDescription.builder()
+                        .ok(true)
+                        .state(ServerConnectionState.CONNECTED)
+                        .type(ServerType.LOAD_BALANCER)
+                        .address(host)
+                        .build()),
+                settings, serverFactory.getSettings());
+        server = serverFactory.create(settings.getHosts().get(0), event -> { }, createServerListener(serverFactory.getSettings()),
+                clusterClock);
+
+        clusterListener.clusterOpening(new ClusterOpeningEvent(clusterId));
+
+        ClusterDescription startingDescription = new ClusterDescription(settings.getMode(), ClusterType.UNKNOWN, Collections.emptyList(),
+                settings, serverFactory.getSettings());
+        ClusterDescription initialDescription = new ClusterDescription(settings.getMode(), ClusterType.LOAD_BALANCED,
+                singletonList(ServerDescription.builder().address(settings.getHosts().get(0)).state(CONNECTING).build()),
+                settings, serverFactory.getSettings());
+        clusterListener.clusterDescriptionChanged(new ClusterDescriptionChangedEvent(clusterId, initialDescription, startingDescription));
+        clusterListener.clusterDescriptionChanged(new ClusterDescriptionChangedEvent(clusterId, description, initialDescription));
+    }
+
+    @Override
+    public ClusterSettings getSettings() {
+        isTrue("open", !isClosed());
+        return settings;
+    }
+
+    @Override
+    public ClusterDescription getDescription() {
+        isTrue("open", !isClosed());
+        waitForSrv();
+        return description;
+    }
+
+    @Override
+    public ClusterId getClusterId() {
+        return clusterId;
+    }
+
+    @Override
+    public ClusterableServer getServer(final ServerAddress serverAddress) {
+        waitForSrv();
+        return server;
+    }
+
+    @Override
+    public ClusterDescription getCurrentDescription() {
+        isTrue("open", !isClosed());
+        return description;
+    }
+
+    @Override
+    public BsonTimestamp getClusterTime() {
+        isTrue("open", !isClosed());
+        return clusterClock.getClusterTime();
+    }
+
+    @Override
+    public ServerTuple selectServer(final ServerSelector serverSelector) {
+        isTrue("open", !isClosed());
+        waitForSrv();
+        if (srvRecordResolvedToMultipleHosts) {
+            throw new MongoClientException("In load balancing mode, the host must resolve to a single SRV record, but instead it resolved "
+                    + "to multiple hosts");
+        }
+        return new ServerTuple(server, description.getServerDescriptions().get(0));
+    }
+
+    private void waitForSrv() {
```

Comment:
SRV support complicated this class a bit.  Once SRV discovery is complete, instances of this class become immutable.  But we have to account for requests to select a server prior to SRV discovery.